src/test/java/io/github/tcdl/msb/mock/adapterfactory/SafeTestMsbConsumerAdapter.java

```diff
@@ -26,6 +27,11 @@ public void pushTestMessage(String jsonMessage) {
         rawMessageHandlers.forEach(handler -> handler.onMessage(jsonMessage, ackHandler));
     }
 
+    @Override
+    public Optional<Long> messageCount() {
+        return Optional.of(10L);
```

Comment:
This will potentially cause trouble in tests and will be very hard to track down. I would create a static variable on the adapter class to be able to manipulate this during tests and set it to a default value of 0L.